cider/util/Datum.h

```diff
@@ -43,7 +43,7 @@ struct VarlenDatum {
       : length(l), pointer(p), is_null(n) {}
 };
 
-union Datum {
+struct Datum {
```

Comment:
why change to struct? I feel this will break `Datum` semantic

Reply:
just try to keep old(int64)/new(int32) date constant value semantics.  Like union semantic will erase the constant type information since different fields will get the same value in union.